05.ls/ls.rb

```diff
@@ -3,22 +3,72 @@
 
 MAXIMUM_COLUMN = 3
 require 'optparse'
+require 'etc'
 
 def main
-  params = ARGV.getopts('r')
-  dirs = params['r'] ? Dir.glob('*').reverse : Dir.glob('*')
+  params = ARGV.getopts('l')
+  dirs = Dir.glob('*')
 
-  files = []
-  total_number_of_files = dirs.size
-  number_of_lines = (total_number_of_files.to_f / MAXIMUM_COLUMN).ceil(0)
+  if params['l']
+    array_of_hard_links = []
+    array_of_file_sizes = []
 
-  slice_the_file(dirs, number_of_lines, files)
+    show_the_total_number_of_blocks(dirs)
 
-  align_the_number_of_elements(files, total_number_of_files)
+    putting_file_details_into_an_array(dirs, array_of_hard_links, array_of_file_sizes)
 
-  sorted_files = files.transpose
+    show_file_details(dirs, array_of_hard_links, array_of_file_sizes)
 
-  show_files(dirs, sorted_files)
+  else
+    files = []
+    total_number_of_files = dirs.size
+    number_of_lines = (total_number_of_files.to_f / MAXIMUM_COLUMN).ceil(0)
+
+    slice_the_file(dirs, number_of_lines, files)
+
+    align_the_number_of_elements(files, total_number_of_files)
+
+    sorted_files = files.transpose
+
+    show_files(dirs, sorted_files)
+  end
+end
+
+def show_the_total_number_of_blocks(dirs)
+  total_of_blocks = 0
+  dirs.each do |dir|
+    fs = File.lstat(dir)
+    total_of_blocks += fs.blocks
+  end
+  puts "total #{total_of_blocks}"
+end
+
+def putting_file_details_into_an_array(dirs, array_of_hard_links, array_of_file_sizes)
+  dirs.each do |dir|
+    fs = File.lstat(dir)
+    array_of_hard_links << fs.nlink.to_s
+    array_of_file_sizes << fs.size.to_s
+  end
+end
+
+def show_file_details(dirs, array_of_hard_links, array_of_file_sizes)
+  file_type = { 'fifo' => 'p', 'characterSpecial' => 'c', 'directory' => 'd', 'blockSpecial' => 'b', 'file' => '-', 'link' => 'l', 'socket' => 's' }
```

Comment:
ちょっと見づらいのでこんな感じで書くと良いと思います。

```suggestion
  file_type = {
    'fifo' => 'p',
    'characterSpecial' => 'c',
    'directory' => 'd',
    'blockSpecial' => 'b',
    'file' => '-',
    'link' => 'l',
    'socket' => 's' 
  }
```